### PR TITLE
Mix-LN: Mixed LayerNorm with speaker-conditioned mixup for acoustic encoder

### PR DESCRIPTION
**Pull request by yqzhishen**
## Summary

Introduces **Mixed LayerNorm (Mix-LN)** — a conditional layer normalization module with mixup-style batch interpolation, integrated into the acoustic encoder. When enabled, speaker embeddings condition the LayerNorm affine parameters, and a Beta-distribution mixup across batch samples improves training stability and generalization.

### `modules/commons/common_layers.py`
- Add `Mixed_LayerNorm` module:
  - Applies standard `F.layer_norm` (weight/bias-free), then conditions affine parameters (scale `gamma`, shift `beta`) on an auxiliary input via a learnable linear projection.
  - During training (batch size > 1), mixes affine parameters between random pairs of samples using Beta-distribution coefficients — a form of manifold mixup at the normalization level.
  - At inference or batch size 1, falls back to standard conditioned normalization.
- Integrate into `EncSALayer`: when `layer_idx` is in the configured `mix_ln_layer` list, replace both pre-attention and pre-FFN `LayerNorm` with `Mixed_LayerNorm`.
- Whitespace cleanup (trailing spaces).

### `modules/fastspeech/acoustic_encoder.py`
- Read `use_mix_ln` and `mix_ln_layer` from hparams and thread through to `FastSpeech2Encoder`.
- Move speaker embedding computation earlier in `forward()` so it can be passed as the conditioning signal to encoder layers.

### `modules/fastspeech/tts_modules.py`
- Thread `layer_idx` and `mix_ln_layer` through `TransformerEncoderLayer` and `FastSpeech2Encoder`.
- Pass `spk_embed` as `cond` kwarg to each encoder layer's forward.

### `deployment/modules/fastspeech2.py`
- Add `uniform_attention_pooling()` helper to downsample frame-level speaker embeddings to phoneme level for ONNX export.
- Support Mix-LN conditioning path during ONNX export: handles both frozen speaker embedding and dynamic speaker embedding cases.

### Configs
- Add `use_mix_ln: false` and `mix_ln_layer: [0, 2]` to `configs/acoustic.yaml` (default off).
- Add `use_mix_ln: true` and `mix_ln_layer: [0, 2]` to `configs/templates/config_acoustic.yaml` (template default on).

## Usage

```yaml
use_mix_ln: true          # enable Mixed LayerNorm
mix_ln_layer: [0, 2]      # which encoder layers use Mix-LN (0-indexed)
use_spk_id: true          # speaker embedding required for conditioning
```

When `use_mix_ln` is false or `mix_ln_layer` is empty, behavior is identical to before — standard `LayerNorm` is used in all encoder layers.

### CODE REVIEW

**Comment by autumn-2-net** [comment]
@codex review

File: modules/fastspeech/acoustic_encoder.py [hunk 1]
```diff
@@ -131,7 +143,7 @@ def forward(
             extra_embed = dur_embed + lang_embed
         else:
             extra_embed = dur_embed
-        encoder_out = self.encoder(txt_embed, extra_embed, txt_tokens == 0)
+        encoder_out = self.encoder(txt_embed, extra_embed, txt_tokens == 0, spk_embed)
```
[the remark this answers is not in the document]
Comment (yqzhishen) [reply modules/fastspeech/acoustic_encoder.py:146]: This code path is never hit with frame-level speaker mixes:

- **Training**: `spk_mix_embed` is not provided — the model uses `spk_embed_id` to look up `[B, 1, H]`, which broadcasts fine with `[B, T_ph, H]` in all cases.
- **Inference**: PyTorch inference goes through the ONNX path (`FastSpeech2AcousticONNX.forward`), which already pools frame-level embeddings to phoneme level via `uniform_attention_pooling`.

The training `FastSpeech2Acoustic.forward` is never called with a frame-level speaker mix, so no shape mismatch occurs in practice.
